docs/howto/migrate/migration-guide.md

```diff
@@ -0,0 +1,591 @@
+val app: ZIO[zio.ZEnv, Nothing, Unit] = 
+  myApp.provideCustomLayer(
+    ((Logging.live ++ Database.live ++ (Logging.live >>> BlobStorage.live)) >>> DocRepo.live) ++
+      ((Logging.live ++ Database.live) >>> UserRepo.live)
+  )
+```
+
+In ZIO 2.x, the `ZIO#injectCustom` does the similar but for the injection:
+
+```scala mdoc:silent:nest
+val app: ZIO[zio.ZEnv, Nothing, Unit] =
+  myApp.injectCustom(
+    Logging.live,
+    DocRepo.live,
+    Database.live,
+    BlobStorage.live,
+    UserRepo.live
+  )
+```
+
+> Note:
+> All `provide*` methods are not deprecated, and they are still necessary and useful for low-level and custom cases. But, in ZIO 2.x, in most cases, it's easier to use `inject`/`wire` methods.
+
+
+| ZIO 1.x and 2.x (manually)                    | ZIO 2.x (automatically) |
+|-----------------------------------------------|-------------------------|
+| `ZIO#provide`                                 | `ZIO#inject`            |
+| `ZIO#provideSomeLayer`                        | `ZIO#injectSome`        |
+| `ZIO#provideCustomLayer`                      | `ZIO#injectCustom`      |
+| Composing manually using `ZLayer` combinators | `ZLayer#wire`           |
+| Composing manually using `ZLayer` combinators | `ZLayer#wireSome`       |
+
+## ZIO Streams
+TODO
+
+## ZIO Services
+
+There are two significant changes in ZIO Services:
+
+| ZIO 1.x                  | ZIO 2.x                 |
+|--------------------------|-------------------------|
+| zio.Clock.Service.live   | zio.Clock.ClockLive     |
+| zio.Console.Service.live | zio.Console.ConsoleLive |
+| zio.System.Service.live  | zio.System.SystemLive   |
+| zio.Random.Service.live  | zio.Random.RandomLive   |
+
+1. All ZIO services moved to the `zio` package:
+
+    | ZIO 1.x               | ZIO 2.x                       |
+    |-----------------------|-------------------------------|
+    | zio.blocking.Blocking | [Removed](#blocking-service)  |
+    | zio.clock.Clock       | zio.Clock                     |
+    | zio.console.Console   | zio.Console                   |
+    | zio.random.Random     | zio.Random                    |
+    | zio.system.System     | zio.System                    |
+
+    And their live implementations renamed and moved to a new path:
+
+    | ZIO 1.x                  | ZIO 2.x                 |
+    |--------------------------|-------------------------|
+    | zio.Clock.Service.live   | zio.Clock.ClockLive     |
+    | zio.Console.Service.live | zio.Console.ConsoleLive |
+    | zio.System.Service.live  | zio.System.SystemLive   |
+    | zio.Random.Service.live  | zio.Random.RandomLive   |
+
+
+2. In ZIO 2.0 all type aliases like `type Logging = Has[Logging.Service]` removed. So we should explicitly use `Has` wrappers when we want to specify dependencies on ZIO services.
+
+So instead of writing `ZLayer[Console with Clock, Nothing, ConsoleLogger]`, we should write `ZLayer[Has[Console] with Has[Clock], Nothing, Has[ConsoleLogger]]`.
```

Comment:
Might also be worth mentioning that instead of things like `ZIO.service[Console.Service]`, we also now do `ZIO.service[Console]`.

Reply:
+1